packager/signer/signer.go

```diff
@@ -288,13 +306,16 @@ func (this *Signer) serveSignedExchange(resp http.ResponseWriter, fetchResp *htt
 
 	// Perform local transformations.
 	r := getTransformerRequest(this.rtvCache, string(fetchBody), signURL.String())
-	transformed, _, err := transformer.Process(r)
+	transformed, metadata, err := transformer.Process(r)
 	if err != nil {
 		log.Println("Not packaging due to transformer error:", err)
 		proxy(resp, fetchResp, fetchBody)
 		return
 	}
 	fetchResp.Header.Set("Content-Length", strconv.Itoa(len(transformed)))
+	if linkHeader := formatLinkHeader(metadata.Preloads); linkHeader != "" {
```

Comment:
should there be some sanitizing/verifying of what to expect within metadata.Preloads here or within formatLinkHeader? it seems we're expecting Preloads to either be defined or empty and with a .Url (if not will fail parsing) and also a .As but we don't check for that to be the case.

what would happen if there is a preload.Url but not a preload.As?

Reply:
OK, I added a basic check for As. I don't think there are any other checks to do? This comes from transformer.Process() anyway, so it shouldn't be structurally wrong.